JSON_serializer/Tests/check_valid_test.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+from unittest import mock
+from requests import RequestException, ConnectionError, Timeout, TooManyRedirects
+
+from JSON_serializer import main
+
+
+class TestValid:
+
+    def test_is_valid_file(self):
+        inp = main.is_valid_file('/home/anastasiya/Downloads/ironman.txt')
+        out = True
+        assert inp == out
+
+    def test_is_valid_file_err(self):
+        inp = main.is_valid_file('ironman.txt')
+        out = False
+        assert inp == out
+
+    def test_open_file(self):
+        with mock.patch('JSON_serializer.main.open') as opened:
+            opened.side_effect = OSError
+            with pytest.raises(OSError):
+                main.file_argument('blackpanther.txt')
+
+    def test_open_file_permission(self):
+        with mock.patch('JSON_serializer.main.open') as opened:
+            opened.side_effect = PermissionError
+            with pytest.raises(PermissionError):
+                main.file_argument('/home/nickfury/work/The Avengers Initiative/characteristics/falcon.txt')
+
+    def test_valid_path(self):
+        with mock.patch('os.path.exists') as path:
+            path.return_value = False
+            assert main.is_valid_file('/home/nickfury/work/The Avengers Initiative/characteristics/hulk.txt') == False
+
+    def test_is_url_err(self):
+        inp = main.is_url('https://www.harrypotter.com')
+        out = False
+        assert inp == out
+
+    @mock.patch.object(main.requests, "get", side_effect=Timeout)
+    def test_is_url_timeout(self, mock_requests_get='https://peggycarter.org'):
+        with pytest.raises(Timeout):
+            main.requests.get(mock_requests_get)
```

Comment:
you shouldn't call `main.requests.get` here. You should execute your app here instead.
Something like this:

- Mock main.requests.get
- Call your app (inside it `requests.get` should be used
- Check the result based on the mocking

Reply:
Will rewrite it.

Reply:
Corrected (attempt no.2)